sha-crypt/src/b64.rs

```diff
@@ -67,8 +85,25 @@ mod tests {
             0x5f, 0xaf, 0x1a, 0xe5, 0x08, 0xe7, 0x7d, 0xd4,
         ];
 
-        let e = super::encode(&original);
-        let d = super::decode(&e).unwrap();
+        let e = super::encode_sha512(&original);
+        let d = super::decode_sha512(&e).unwrap();
+
+        for i in 0..d.len() {
+            assert_eq!(&original[i], &d[i]);
+        }
+    }
+
+    #[cfg(feature = "simple")]
+    #[test]
+    fn test_encode_decode_sha256() {
+        let original: [u8; 32] = [
+            0x0b, 0x5b, 0xdf, 0x7d, 0x92, 0xe2, 0xfc, 0xbd, 0xab, 0x57, 0xcb, 0xf3, 0xe0, 0x03,
+            0x16, 0x62, 0xd3, 0x6e, 0xa0, 0x57, 0x44, 0x8c, 0xca, 0x35, 0xec, 0x80, 0x75, 0x2a,
+            0x5f, 0xaf, 0x1a, 0xe5
+        ];
+
+        let e = super::encode_sha256(&original);
+        let d = super::decode_sha256(&e).unwrap();
```

Comment:
This test also fails.

Note, that when doing another roundtrip (encode +  decode), it converges instantly.